src/app/GitUI/LeftPanel/RepoObjectsTree.ContextActions.cs

```diff
@@ -289,5 +299,31 @@ public TMenuItem CreateMenuItem<TMenuItem, TNode>(Action<TNode> onClick, Transla
             RegisterClick(result, onClick);
             return result;
         }
+
+        private void EnableFavoritesContextMenu(bool hasSingleSelection, NodeBase? selectedNode)
+        {
+            if (!hasSingleSelection || selectedNode is not (LocalBranchNode or RemoteBranchNode))
+            {
+                EnableMenuItems(false, mnubtnRemoveFromFavorites, mnubtnAddToFavorites);
+                return;
+            }
+
+            bool isFavorite = selectedNode?.TreeViewNode?.FullPath?.Split('/').FirstOrDefault()?.Contains(TranslatedStrings.Favorites) ?? false;
+
+            EnableMenuItems(isFavorite, mnubtnRemoveFromFavorites);
+            EnableMenuItems(!isFavorite, mnubtnAddToFavorites);
+        }
+
+        private void AddToFavorites(NodeBase node)
+        {
+            // Implementation for adding to favorites
+            _favoritesTree.Add(node);
+        }
+
+        private void RemoveFromFavorites(NodeBase node)
+        {
+            // Implementation for removing from favorites
+            _favoritesTree.Remove(node);
+        }
```

Comment:
The separation was intended to allow for additional checks, but since no specific validations are needed, inlining them at the call sites would be fine. I’ll make this adjustment to simplify the code. Thanks for the suggestion!